app/extractors/BeneficiaryExtractor.scala

```diff
@@ -79,6 +113,36 @@ trait BeneficiaryExtractor[T <: Beneficiary] {
     }
   }
 
+  def extractDateOfBirth(dateOfBirth: Option[LocalDate], answers: UserAnswers) : Try[UserAnswers] = {
+    dateOfBirth match {
+      case Some(dob) =>
+        answers.set(dateOfBirthYesNoPage, true)
+          .flatMap(_.set(dateOfBirthPage, dob))
+      case None =>
+        // Assumption that user answered no as dob is not provided
+        answers.set(dateOfBirthYesNoPage, false)
+    }
+  }
```

Comment:
Likewise I would keep this in the `IndividualBeneficiaryExtractor`

Reply:
Updated